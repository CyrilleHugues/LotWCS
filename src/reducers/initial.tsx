import { makeTypedFactory } from 'typed-immutable-record';
import * as Immutable from 'immutable';
import { IStoreStateJS, IStoreState } from '../types/state';

import { resetToInitialState } from '../actions/initial';
import { IDataSkill } from '../types/skills';
import { skills } from '../data/skills';

const defaultState: IStoreStateJS = {
  name: 'No Name',
  concept: 'No Concept',
  archetype: '',
  archetypeModified: false,
  rank: '',
  rankModified: false,
  skills: Immutable.List(skills.map((s: IDataSkill) => {
    return {
      name: s.name,
      value: 0,
      specialities: Immutable.List<string>(),
    };
  })),
  history: Immutable.List([resetToInitialState()]),
};

export const initialStateFactory = makeTypedFactory<IStoreStateJS, IStoreState>(defaultState);
