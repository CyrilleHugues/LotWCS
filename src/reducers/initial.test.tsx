import { initialStateFactory } from './initial';

it('initialStateFactory', () => {
  expect( initialStateFactory() ).toMatchSnapshot();
});
