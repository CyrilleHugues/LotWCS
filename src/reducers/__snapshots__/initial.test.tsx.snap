// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialStateFactory 1`] = `
Object {
  "archetype": "",
  "archetypeModified": false,
  "concept": "No Concept",
  "history": Array [
    Object {
      "type": "INITIAL_STATE",
    },
  ],
  "name": "No Name",
  "rank": "",
  "rankModified": false,
  "skills": Array [
    Object {
      "name": "Awareness",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Confidence",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Crafting",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Finesse",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Hardiness",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Inspire",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Learning",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Medicine",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Might",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Perform",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Politics",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Ride",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Stealth",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Survival",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Tactics",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Wu Wei",
      "specialities": Immutable.List [
],
      "value": 0,
    },
  ],
}
`;
